src/formatting/comment.rs

```diff
@@ -948,25 +948,30 @@ fn light_rewrite_comment(
     config: &Config,
     is_doc_comment: bool,
 ) -> String {
-    let lines: Vec<&str> = orig
+    let lines: Vec<String> = orig
         .lines()
         .map(|l| {
             // This is basically just l.trim(), but in the case that a line starts
             // with `*` we want to leave one space before it, so it aligns with the
             // `*` in `/*`.
             let first_non_whitespace = l.find(|c| !char::is_whitespace(c));
-            let left_trimmed = if let Some(fnw) = first_non_whitespace {
-                if l.as_bytes()[fnw] == b'*' && fnw > 0 {
-                    &l[fnw - 1..]
+            let (blank, left_trimmed) = if let Some(fnw) = first_non_whitespace {
+                if l.as_bytes()[fnw] == b'*' {
+                    // Ensure '*' is preceeded by blank and not by a tab.
+                    (" ", &l[fnw..])
                 } else {
-                    &l[fnw..]
+                    ("", &l[fnw..])
                 }
             } else {
-                ""
+                ("", "")
             };
             // Preserve markdown's double-space line break syntax in doc comment.
-            trim_end_unless_two_whitespaces(left_trimmed, is_doc_comment)
+            (
+                blank,
+                trim_end_unless_two_whitespaces(left_trimmed, is_doc_comment),
+            )
         })
+        .map(|(b, l)| format!("{}{}", b, l))
```

Comment:
Good catch! I don't see a need for another `map` call though, let's just return the allocated string in the closure in the previous block like was done before

```suggestion
            format!(
                "{}{}",
                blank,
                trim_end_unless_two_whitespaces(left_trimmed, is_doc_comment),
            )
        })
```

Reply:
Done